modules/custom-status/custom-status.php

```diff
@@ -459,747 +466,806 @@ function post_admin_header() {
+			$name = isset( $_POST['name'] ) ? sanitize_text_field( trim( $_POST['name'] ) ) : '';
+			$description = isset( $_POST['description'] ) ? stripslashes( wp_filter_nohtml_kses( trim( $_POST['description'] ) ) ) : '';
+
+			/**
+			 * Form validation for editing custom status
+			 *
+			 * Details
+			 * - 'name' is a required field and can't conflict with existing name or slug
+			 * - 'description' is optional
+			 */
+			$_REQUEST['form-errors'] = array();
+			// Check if name field was filled in
+			if ( empty( $name ) ) {
+				$_REQUEST['form-errors']['name'] = __( 'Please enter a name for the status', 'edit-flow' );
+			}
+			// Check that the name isn't numeric
+			if ( is_numeric( $name ) ) {
+				$_REQUEST['form-errors']['name'] = __( 'Please enter a valid, non-numeric name for the status.', 'edit-flow' );
+			}
+			// Check that the status name doesn't exceed 20 chars
+			if ( strlen( $name ) > 20 ) {
+				$_REQUEST['form-errors']['name'] = __( 'Status name cannot exceed 20 characters. Please try a shorter name.', 'edit-flow' );
+			}
+			// Check to make sure the status doesn't already exist as another term because otherwise we'd get a weird slug
+			$term_exists = term_exists( sanitize_title( $name ), self::taxonomy_key );
+			if ( $term_exists && isset( $term_exists['term_id'] ) && $term_exists['term_id'] != $existing_status->term_id ) {
+				$_REQUEST['form-errors']['name'] = __( 'Status name conflicts with existing term. Please choose another.', 'edit-flow' );
+			}
+			// Check to make sure the status doesn't already exist
+			$search_status = $this->get_custom_status_by( 'slug', sanitize_title( $name ) );
+			if ( $search_status && $search_status->term_id != $existing_status->term_id ) {
+				$_REQUEST['form-errors']['name'] = __( 'Status name conflicts with existing status. Please choose another.', 'edit-flow' );
+			}
+			// Check to make sure the name is not restricted
+			if ( $this->is_restricted_status( strtolower( sanitize_title( $name ) ) ) ) {
+				$_REQUEST['form-errors']['name'] = __( 'Status name is restricted. Please choose another name.', 'edit-flow' );
+			}
 
-		if ( !current_user_can( 'manage_options' ) )
-			wp_die( $this->module->messages['invalid-permissions'] );
+			// Kick out if there are any errors
+			// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotValidated
+			if ( count( $_REQUEST['form-errors'] ) ) {
+				$_REQUEST['error'] = 'form-error';
+				return;
+			}
 
-		if ( !$existing_status = $this->get_custom_status_by( 'id', (int)$_GET['term-id'] ) )
-			wp_die( $this->module->messages['status-missing'] );
+			// Try to add the new post status
+			$args = array(
+				'name' => $name,
+				'slug' => sanitize_title( $name ),
+				'description' => $description,
+			);
+			$return = $this->update_custom_status( $existing_status->term_id, $args );
+			if ( is_wp_error( $return ) ) {
+				wp_die( esc_textarea( __( 'Error updating post status.', 'edit-flow' ) ) );
```

Comment:
`esc_html()` should be fine and more restrictive here.

Reply:
Or, even better

```suggestion
				wp_die( esc_html__( 'Error updating post status.', 'edit-flow' ) );
```

Reply:
Done